distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/JobManagerPaths.java

```diff
@@ -3,88 +3,100 @@
+    private static final String ASSIGNED = "assigned";
+    private static final String AVAILABLE = "available";
+    private static final String WORK_POOLED_JOB_ID = "work-pooled";
+    private static final String WORK_POOL = "work-pool";
 
     final String rootPath;
 
     JobManagerPaths(String rootPath) {
         this.rootPath = rootPath;
     }
 
-    String getAssignmentVersion() {
-        return ZKPaths.makePath(rootPath, ASSIGNMENT_VERSION);
-    }
-
-    String getRegistrationVersion() {
-        return ZKPaths.makePath(rootPath, REGISTRATION_VERSION);
-    }
-
-    String getWorkersPath() {
-        return ZKPaths.makePath(rootPath, "workers");
+    public String getWorkersAlivePath() {
+        return path(ALIVE);
     }
 
     String getWorkerAliveFlagPath(String workerId) {
-        return ZKPaths.makePath(getWorkersAlivePath(), workerId);
+        return path(ALIVE, workerId);
     }
 
-    String getWorkerPath(String workerId) {
-        return ZKPaths.makePath(getWorkersPath(), workerId);
+    String getAssignmentVersion() {
+        return path(ASSIGNMENT_VERSION);
     }
 
-    String getWorkerAvailableJobsPath(String worker) {
-        return ZKPaths.makePath(getWorkerPath(worker), "available");
+    String getLeaderLatchPath() {
+        return path(LEADER_LATCH);
     }
 
-    String getWorkerAssignedJobsPath(String worker) {
-        return ZKPaths.makePath(getWorkerPath(worker), "assigned");
+    String getWorkPooledLocksPath() {
+        return path(LOCKS, WORK_POOLED_JOB_ID);
     }
 
-    String getLeaderLatchPath() {
-        return ZKPaths.makePath(rootPath, "leader-latch");
+    String getWorkItemLock(String jobId, String workItem) {
+        return path(LOCKS, WORK_POOLED_JOB_ID, jobId,
+                String.format("work-share-%s.lock", workItem));
     }
 
-    public String getWorkersAlivePath() {
-        return ZKPaths.makePath(rootPath, "alive");
+    String getRegistrationVersion() {
+        return path(REGISTRATION_VERSION);
     }
 
-    String getAvailableWorkPooledJobPath(String workerId) {
-        return ZKPaths.makePath(getWorkerPath(workerId), "available", WORK_POOLED_JOB_ID);
+    String getWorkersPath() {
+        return path(WORKERS);
     }
 
-    String getAvailableWorkPoolPath(String workerId, String jobId) {
-        return ZKPaths.makePath(getAvailableWorkPooledJobPath(workerId), jobId, WORK_POOL);
+    String getWorkerPath(String workerId) {
+        return path(WORKERS, workerId);
     }
 
-    String getAssignedWorkPoolPath(String workerId, String jobId) {
-        return ZKPaths.makePath(getAssignedWorkPooledJobsPath(workerId), jobId, WORK_POOL);
+    String getWorkerAssignedJobsPath(String workerId) {
+        return path(WORKERS, workerId, ASSIGNED);
     }
 
     String getAssignedWorkPooledJobsPath(String workerId) {
-        return ZKPaths.makePath(getWorkersPath(), workerId, "assigned", WORK_POOLED_JOB_ID);
+        return path(WORKERS, workerId, ASSIGNED, WORK_POOLED_JOB_ID);
```

Comment:
same suggestion, convert: 
/workers/3/assigned/work-pooled/my-job-id/my-work-item-1
to
/workers/3/assigned/my-job-id/my-work-item-1